libs/execution_context/ebpf_maps.c

```diff
@@ -921,12 +917,14 @@ _reap_oldest_map_entry(_In_ ebpf_core_map_t* map)
 
     // Walk through all the keys and values and find the oldest one.
     for (;;) {
+        uint64_t* key_age = NULL;
         result = ebpf_hash_table_next_key_pointer_and_value(
-            lru_map->key_history, previous_key, &next_key, (uint8_t**)&key_age);
+            (ebpf_hash_table_t*)lru_map->core_map.data, previous_key, &next_key, (uint8_t**)&value);
         if (result != EBPF_SUCCESS) {
```

Comment:
should we add a check before the loop to see if the LRU is empty or partially filled and return 'True' if so?  reaping is really a 'no-op' in such cases so why bother proceeding?  Also, the error returned from within the loop would then become more meaningful.

To be clear, the current code calls this function only if the LRU is full.  But newer code might find some 'creative' uses for this function :-)

This is just a thought.  I'm fine with this implementation.

Reply:
This block of code is going to be replaced in the next PR. In addition to what you pointed out, there is also the issue that after reaping one entry, the insert can still fail if another thread inserts before this one can.